interfaces/SolARGenericSemanticSegmentationOpencv.h

```diff
@@ -59,7 +59,7 @@ class SOLAROPENCV_EXPORT_API SolARGenericSemanticSegmentationOpencv : public org
     /// @param[in] image The input image.
     /// @param[out] mask The mask has same size as the input image, in which the value of each pixel is corresponding to the class id.
     /// @return FrameworkReturnCode::_SUCCESS if the segmentation succeed, else FrameworkReturnCode::_ERROR_
-    FrameworkReturnCode segment(const SRef<SolAR::datastructure::Image> image, SRef<SolAR::datastructure::Image> &mask) override;
+    FrameworkReturnCode segment(const SRef<SolAR::datastructure::Image> image, SRef<SolAR::datastructure::Image> mask) override;
```

Comment:
I know this is done in many places, so I'm just putting it here, feel free to do this in this PR or later.
To express Image is not modified (which I assume, is what we want here):
`SRef<const SolAR::datastructure::Image> image`

Reply:
If we want to instantiate the image within the segment method, we need to pass the SRef<image> in reference. This is useful if we want to do:
`SRef<Image> segmentedImage; 
segment(inputImage, segmentedImage);`

Reply:
Regarding JM's comment, yes, I think that in ISemanticSegmentation.h we use const SRef<SolAR::datastructure::Image> image is intended to make sure that inside segment() image will not be modified.  You mean the right syntax is to use SRef<const Image> instead of const SRef<Image>?

Reply:
I agree with Jerome that for the mask (or segmentedImage) we should pass the SRef in reference since mask (or segmentedImage) will be created inside the segment() method

Reply:
the & has been put back before mask